File: app/src/main/kotlin/com/mozzid/domain/model/ClassificationResult.kt
package com.mozzid.domain.model

/**
 * Output of a [com.mozzid.domain.classifier.SpeciesClassifier] run over one
 * wingbeat sample: the best match plus the runner-up and the measured frequency.
 * Pure data — no persistence.
 */
data class ClassificationResult(
    val primary: Species,
    val runner: Species,
    val confidence: Int,       // 0..100
    val runnerConfidence: Int, // 0..100
    val wingbeatHz: Int,       // measured fundamental, e.g. 612
)

File: reference/flutter-src/lib/domain/models/classification_result.dart
import 'species.dart';

/// Output of a [SpeciesClassifier] run over one wingbeat sample: the best match
/// plus the runner-up and the measured frequency. Pure data — no persistence.
class ClassificationResult {
  const ClassificationResult({
    required this.primary,
    required this.runner,
    required this.confidence,
    required this.runnerConfidence,
    required this.wingbeatHz,
  });

  final Species primary;
  final Species runner;

  /// Primary confidence, 0–100.
  final int confidence;

  /// Runner-up confidence, 0–100.
  final int runnerConfidence;

  /// Measured wingbeat fundamental in Hz (e.g. 612).
  final int wingbeatHz;
}
